include/SGM.hpp

```diff
@@ -1,13 +1,94 @@
 #pragma once
+#include <EVT/io/ADC.hpp>
+#include <EVT/io/CANDevice.hpp>
+#include <EVT/io/CANOpenMacros.hpp>
+#include <co_core.h>
+#include <dev/StrainGauge.hpp>
+
+#define NUM_GAUGES 4
+
+namespace IO = EVT::core::IO;
 
 namespace SGM {
 
 /**
- * This is an example of a class for a board
+ * Measures strain/stress on component using reading from ADC.
  */
-class SGM {
+class SGM : public CANDevice {
 public:
+    /**
+     * Constructor takes an array of 4 strain gauges
+     *
+     * @param[in] gauges the strain gauges
+     */
+    explicit SGM(DEV::StrainGauge gauges[NUM_GAUGES]);
```

Comment:
You'll want to pass this as an array of pointers, so you don't have to copy all the objects. You may also be able to pass these by reference